src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,597 @@
+                      std::is_same<T, XYZ_RGBA_DATA>::value ||
+                      std::is_same<T, XYZ_I_DATA>::value ||
+                      std::is_same<T, INTEREST_POINT_XYZ_DATA>::value ||
+                      std::is_same<T, XYZ_NORMAL_DATA>::value ||
+                      std::is_same<T, XYZ_NORMAL_RGBA_DATA>::value, "yarp::sig::PointCloud<T>: T chosen is not supported");
+    public:
+
+        /**
+         * @brief PointCloud, default constructor.
+         */
+        PointCloud()
+        {
+            data.clear();
+            setPointType();
+        }
+
+        /**
+         * @brief PointCloud, copy constructor.
+         * Clones the content of another point cloud.
+         * @param alt the point cloud to clone.
+         */
+        template<class T1>
+        PointCloud(const PointCloud<T1>& alt)
+        {
+            setPointType();
+            copy<T1>(alt);
+            return;
+
+        }
+
+        /**
+         * @brief Resize the PointCloud.
+         * @param width.
+         * @param height.
+         */
+        virtual void resize(size_t width, size_t height)
+        {
+            header.width = width;
+            header.height = height;
+            data.resize(width * height);
+        }
+
+        /**
+         * @brief Resize the PointCloud.
+         * @param width.
+         * Calling this function the cloud will become NOT organizedS
+         */
+        virtual void resize(size_t width)
+        {
+            header.width = width;
+            header.height = 1;
+            data.resize(width);
+        }
+
+        /**
+         * @brief Get the pointer to the data.
+         * @return the pointer to the data.
+         */
+        virtual const char* getRawData() const override
+        {
+            return data.getMemoryBlock();
+        }
+
+        /**
+         * @brief Get the size of the data + the header in terms of
+         * number of bytes.
+         * @return the size of the data sent through the network
+         */
+        virtual size_t wireSizeBytes() const override
+        {
+            return sizeof(header) + dataSizeBytes();
+        }
+
+        /**
+         * @brief Get the size of the data in terms of
+         * number of bytes.
+         * @return the size of the data
+         */
+        virtual size_t dataSizeBytes() const override
+        {
+            return header.width*header.height*(sizeof(T));
+        }
+
+        /**
+         * @brief
+         * @return the number of points of the PointCloud
+         */
+        virtual size_t size() const override
+        {
+            return data.size();
+        }
+
+        /**
+         * @brief Obtain the point given by the (column, row) coordinates.
+         * Only works on organized clouds (those that have height != 1).
+         * @param u, column coordinate
+         * @param v, row coordinate
+         */
+        inline T& operator()(size_t u, size_t v) {
+            yAssert(isOrganized());
```

Comment:
You have already a error handling behaviour (even if not documented) in the following for, why not to use it also for the `isOrganized` case? Crashing is not always a desired behavior on error.

Reply:
```c++
inline const PointT&
at (int column, int row) const
   {
       if (this->height > 1)
           return (points.at (row * this->width + column));
       else
           throw IsNotDenseException ("Can't use 2D indexing with a unorganized point cloud");
  }
```
This is how PCL handles the access through `u` and `v` in a pointcloud that it is NOT organized.
I can add an error message?